### PR TITLE
auto-update the fossgis instances

### PR DESCRIPTION
**Pull request by nilsnolde**
follow-up to https://github.com/valhalla/valhalla/pull/4716. will now try to run the action itself.

### CODE REVIEW

**Review by kevinkreiser — APPROVED** [review]
feel free to merge this if it helps testing!

**Comment by nilsnolde** [comment]
I could manually trigger the workflow from this branch, but thanks. seems to work now. what it does on pushes to master (and only on changes to interesting directories):
- ssh's into each machine with a repo-wide github secret for the private key
- pulls, builds and installs the project on each machine
- on the service machine it re-generates the valhalla config, on the builder machine the forever-looping script is doing that itself

do note, this won't have an immediate effect on either the service or the builder. we don't restart the processes, we wait for the next planet graph build to finish which will automatically restart all processes on both machines so the changes will be picked up then.

from my side, this is good for merging @kevinkreiser, but feel free to look over the 2 whole files (which are also this PR's changes).

**Comment by nilsnolde** [comment]
the script runs as `sudo` because of the installation. I know I could make `/usr/local` allowed for the user, but doesn't really matter IMO, the workflow can't run from a fork in a PR, only maintainers have access to the secrets. but I'm 99% sure that master merges will run regardless if from a fork or not.

File: scripts/update_public_server.sh
Comment (nilsnolde) [inline scripts/update_public_server.sh:15]: ccache is installed

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**
